frontend/packages/ceph-storage-plugin/src/components/modals/add-capacity-modal/add-capacity-modal.tsx

```diff
@@ -200,58 +237,38 @@ export const AddCapacityModal = (props: AddCapacityModalProps) => {
+            )}
           </FormGroup>
-          {isNoProvionerSC ? (
-            <PVsAvailableCapacity
-              replica={replica}
-              data-test-id="ceph-add-capacity-pvs-available-capacity"
-              storageClass={storageClass}
-              data={pvData}
-              loaded={pvLoaded}
-              loadError={pvLoadError}
-            />
-          ) : (
-            <>
-              <FormGroup
-                className="pf-u-py-sm"
-                fieldId="request-size"
-                id="requestSize__FormGroup"
-                label={t('ceph-storage-plugin~Raw Capacity')}
-                labelIcon={<FieldLevelHelp>{requestedCapacityTooltip(t)}</FieldLevelHelp>}
-              >
-                <TextInput
-                  isDisabled
-                  id="request-size"
-                  className={classNames('pf-c-form-control', 'ceph-add-capacity__input')}
-                  type="number"
-                  name="requestSize"
-                  value={osdSizeWithoutUnit}
-                  aria-label="requestSize"
-                  data-test-id="requestSize"
-                />
-                {provisionedCapacity && (
-                  <TextContent className="ceph-add-capacity__provisioned-capacity">
-                    {' '}
-                    {t('ceph-storage-plugin~x {{ replica, number }} replicas =', {
-                      replica,
-                    })}{' '}
-                    <strong data-test="provisioned-capacity">{provisionedCapacity}&nbsp;TiB</strong>
-                  </TextContent>
+          {!!selectedSCName &&
+            (isNoProvionerSC ? (
+              <PVsAvailableCapacity
+                replica={replica}
+                data-test-id="ceph-add-capacity-pvs-available-capacity"
+                storageClass={storageClass}
+                data={pvData}
+                loaded={pvLoaded}
+                loadError={pvLoadError}
+              />
+            ) : (
+              <>
+                {!!osdSizeWithoutUnit && (
+                  <RawCapacity t={t} replica={replica} osdSizeWithoutUnit={osdSizeWithoutUnit} />
                 )}
                 <TextContent className="pf-u-font-weight-bold pf-u-secondary-color-100 ceph-add-capacity__current-capacity">
                   {t('ceph-storage-plugin~Currently Used:')}&nbsp;
                   {currentCapacity}
                 </TextContent>
-              </FormGroup>
-            </>
-          )}
+              </>
+            ))}
         </ModalBody>
         <ModalSubmitFooter
           inProgress={inProgress}
           errorMessage={errorMessage}
           submitText={t('ceph-storage-plugin~Add')}
           cancel={cancel}
-          submitDisabled={isNoProvionerSC && (!availablePvsCount || nodesError)}
+          submitDisabled={isNoProvionerSC ? !availablePvsCount || nodesError : !osdSizeWithoutUnit}
```

Comment:
This will disable adding new storage class
```suggestion
          submitDisabled={isNoProvionerSC ? !availablePvsCount || nodesError : false}
```

Reply:
won't it disable the button for Internal-Mode too ?
If we are expanding internal-mode cluster.

Reply:
Okay, its the first device set.
Looks fine then.

Reply:
It will disable the "Add" button for all prov-sc when we are expanding for Internal-mode.
We only want to disable this button (for internal-mode) when OSD size is not 512GiB/2TiB/4TiB.